src/main/java/com/mongodb/spark/sql/connector/config/AbstractMongoConfig.java

```diff
@@ -225,8 +220,16 @@ public int hashCode() {
     return Objects.hash(options, usageMode);
   }
 
-  Map<String, String> withOverrides(final Map<String, String> overrides) {
+  Map<String, String> withOverrides(final String context, final Map<String, String> overrides) {
```

Comment:
This is where the context for the overrides (eg: `dfw.options(...)` overrides any set configuration).